code/messaging/src/main/java/com/adobe/marketing/mobile/Proposition.java

```diff
@@ -0,0 +1,77 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile;
+
+import java.lang.ref.WeakReference;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A {@link Proposition} object encapsulates offers and the information needed for tracking offer interactions.
+ */
+public class Proposition {
+    // Unique proposition identifier
+    private final String uniqueId;
+    // Scope string
+    private final String scope;
+    // Scope details map
+    private final Map<String, Object> scopeDetails;
+    // List containing proposition decision items
+    private final List<PropositionItem> propositionItems;
+
+    public Proposition(final String uniqueId, final String scope, final Map<String, Object> scopeDetails, final List<PropositionItem> propositionItems) {
+        this.uniqueId = uniqueId;
+        this.scope = scope;
+        this.scopeDetails = scopeDetails;
+        this.propositionItems = propositionItems;
+        if (this.propositionItems != null && !this.propositionItems.isEmpty()) {
+            this.propositionItems.get(0).proposition = new WeakReference<>(this);
```

Comment:
Here you need to iterate on the propositionItems list and populate the parent proposition ref in each if null.
```
for (PropositionItem item: this.propositionItems) {
    if (item.propositionReference == null) {
        item.propositionReference = new WeakReference<>(this);
    }
}
```